src/org/scada_lts/web/mvc/api/EventsAPI.java

```diff
@@ -104,28 +157,194 @@
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Silence all Events Alarm from REST API
+     *
+     * @param request Request containing user data
+     * @return Response
+     */
+    @PostMapping(value = "/silenceAll")
+    public ResponseEntity<String> silenceAllEvents(HttpServletRequest request) {
+        LOG.info("GET::/api/events/silenceAll");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                Date time = new Date();
+                eventService.silenceAll(time.getTime(), user.getId(), 0);
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Acknowledge selected Events Alarm from REST API
+     *
+     * @param request Request containing user data
+     * @return Response
+     */
+    @PostMapping(value = "/ackSelected")
+    public ResponseEntity<String> acknowledgeSelectedEvents(@RequestBody JsonIdSelection query, HttpServletRequest request) {
+        LOG.info("GET::/api/events/ackSelected");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                Date time = new Date();
+                List <Integer> ids = new ArrayList<Integer>();
+                for (String id: query.getIds().split(",")) {
+                    ids.add(Integer.parseInt(id.trim()));
+                }
+                eventService.ackSelected(time.getTime(), user.getId(), 0, ids);
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
+
+    /**
+     * Silence selected Events Alarm from REST API
+     *
+     * @param request Request containing user data
+     * @return Response
+     */
+    @PostMapping(value = "/silenceSelected")
+    public ResponseEntity<String> silenceSelectedEvents(@RequestBody JsonIdSelection query, HttpServletRequest request) {
+        LOG.info("GET::/api/events/silenceSelectedEvents");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                Date time = new Date();
+                List <Integer> ids = new ArrayList<Integer>();
+                for (String id: query.getIds().split(",")) {
+                    ids.add(Integer.parseInt(id.trim()));
+                }
+                eventService.silenceEvents(ids, time.getTime(), user.getId(), 0);
```

Comment:
Remove` Date time` and unnecessary arguments ->
`eventService.silenceEvents(ids, user.getId());`